tests/integration/test_compositions.py

```diff
@@ -58,13 +58,56 @@ def test_looping_embarrassingly_parallel_step(test_specific_results_dir: Path) -
         ).exists()
 
 
+EP_SECTION_MAPPING = {
+    "parallel_step": {
+        "pipeline_spec": "pipeline_parallel_step.yaml",
+        "schema_name": "ep_parallel_step",
+        "node_prefix": "parallel_split",
+    },
+    "loop_step": {
+        "pipeline_spec": "pipeline_loop_step.yaml",
+        "schema_name": "ep_loop_step",
+        "node_prefix": "loop",
+    },
+}
+
+
 @pytest.mark.slow
-def test_embarrassingly_parallel_parallel_step(test_specific_results_dir: Path) -> None:
-    pipeline_specification = EP_SPECIFICATIONS_DIR / "pipeline_parallel_step.yaml"
+@pytest.mark.parametrize(
+    "step_type, expected_counter, num_rows_multiplier",
+    [
+        (
+            "loop_step",
+            2,
+            1,
+        ),
+        (
+            "parallel_step",
+            # We will run step_1 exactly one time on each of the four subsets of data
```

Comment:
"four subsets" could be a bit clearer. There are two MECE subsets (made by the EP), each duplicated 2 times (for the parallel splits).